meerkat_frontend/src/js/technical/tabletSubmission.js

```diff
@@ -211,51 +178,52 @@ function fillTabletData(clinicObj) {
     });
 }
 
-//This method will handle all the data related to week columns "tot and reg"
-function buildWeekData(clinicObj, minuWeek, colKey) {
-    var weekVal = week - minuWeek;
-    $.each(clinicObj.deviceSubmissions, function(index, deviceVal) {
-        $.each(tabletDataArray, function(index, value) {
-            if (value.DevicesIDs === "- DeviceId  " + deviceVal.deviceId) {
-                var itemName = colKey + weekVal;
-                value[itemName] = deviceVal.submissionsCount;
-            }
-        });
-    });
-}
 
+//this method will handle to fill the data from the apis into the table
+function fillAPIResultData(clinicObj, minusNum, colKey) {
 
-//This method will handle all the data related to year columns
-function buildYearData(clinicObj, minuYear, colKey) {
-    var yearVal = current_Year - minuYear;
-    $.each(clinicObj.deviceSubmissions, function(index, deviceVal) {
-        $.each(tabletDataArray, function(index, value) {
-            if (value.DevicesIDs === "- DeviceId  " + deviceVal.deviceId) {
-                var itemName = colKey + yearVal;
-                value[itemName] = deviceVal.submissionsCount;
-            }
+    //This section will handle all the data related to week columns "tot and reg"
+    if (colKey == "casesInWeek" || colKey == "noOfRegistersInWeek") {
+        var weekVal = week - minusNum;
+        $.each(clinicObj.deviceSubmissions, function(index, deviceVal) {
+            $.each(tabletDataArray, function(index, value) {
+                if (value.DevicesIDs === "DeviceId " + deviceVal.deviceId) {
+                    var itemName = colKey + weekVal;
+                    value[itemName] = deviceVal.submissionsCount;
+                }
+            });
         });
-    });
+    } else if (colKey == "casesInYear") {
+        //This section will handle all the data related to year columns
+        var yearVal = current_Year - minusNum;
+        $.each(clinicObj.deviceSubmissions, function(index, deviceVal) {
+            $.each(tabletDataArray, function(index, value) {
+                if (value.DevicesIDs === "DeviceId " + deviceVal.deviceId) {
+                    var itemName = colKey + yearVal;
+                    value[itemName] = deviceVal.submissionsCount;
+                }
+            });
+        });
+    } else if (colKey == "casesInMonthL" || colKey == "casesInMonthC") {
+        //This method will handle all the data related to month columns
```

Comment:
Here "this method" if `else if` section (which is not even a method but a block) and in line 209 "this method" is `getMonth()`. This is very confusing causing more harm then good with this comment.
As said earlier I suggest removing this comment line.

Reply:
edited, some comments i leave it for me so i can remember very fast some cases like 
this comment:
// +1 because this method return index for month jan = 0 & dec =11